JIT.cpp

```diff
@@ -35,9 +38,13 @@ JIT::JIT(orc::JITTargetMachineBuilder machine_builder,
                                          m_mangler(m_execution_session, m_data_layout),
                                          m_ctx(std::make_unique<llvm::LLVMContext>())
 {
+    static bool s_run_guard = false;
```

Comment:
Do you have multiple threads that could all call this function? If so, might consider a mutex.

Reply:
No, this was intended to be single-threaded, but I'll add it just in case.

Reply:
Actually that seems like more trouble than it's worth currently - we'd have to make a lot of stuff thread-safe if we made this thread-safe (like the logging system).